cmlutils/utils.py

```diff
@@ -287,3 +287,29 @@ def _get_runtimes_v2(runtimes, editor="Workbench", edition="Standard"):
     legacy_runtime_image_map["default"] = legacy_runtime_image_map["python3"]
 
     return legacy_runtime_image_map
+
+
+def compare_metadata(import_data, export_data, import_data_list, export_data_list, skip_field=None):
+    if skip_field is None:
+        skip_field = []
+
+    data_list_diff = list(set(sorted(export_data_list)) - set(sorted(import_data_list)))
+    config_differences = {}
+
+    for im_data in import_data:
```

Comment:
Three nested for loops are expensive in terms of time complexity. Can you optimize this code by reducing the nested for loops? May be put second list in dictionary, then loop over first list finding matching element from the dictionary.

Reply:
Done